include/xtensor/xbuilder.hpp

```diff
@@ -135,7 +135,7 @@ namespace xt
             }
         };
 
-        template <class T, template<class, class> class F>
+        template <class T, template <class> class K, class F = K<T>>
```

Comment:
I don't understand why you use a third template parameter with a default value set to `K<T>`. It seems you don't use the `K` parameter elsewhere in this class, so maybe you could just do something like:
```cpp
template <class T, template <class> class F>
class fn_impl
{
    using functor_type = F<T>;
    ....
};
```
Or, even better, add a `value_type` typedef in `eye_fn`, and retrieve it in `fn_impl`. This way you don't need the `T` template parameter.